Make GOOGLE_SERVICE_ACCOUNT_ID optional

zackcl opened the pull request:
This PR makes `GOOGLE_SERVICE_ACCOUNT_ID` optional environment variable.

zackcl commented on backend/packages/Upgrade/src/env.ts:
```diff
@@ -79,7 +79,7 @@ export const env = {
   google: {
     clientId: getOsEnv('GOOGLE_CLIENT_ID'),
-    serviceAccountId: getOsEnv('GOOGLE_SERVICE_ACCOUNT_ID'),
+    serviceAccountId: getOsEnvOptional('GOOGLE_SERVICE_ACCOUNT_ID'),
```
It's not an array. It's a comma-separated string.

danoswaltCL replied:
no it's a string, comma separated for multiple

zackcl commented:
I will try to make both `GOOGLE_CLIENT_ID` and `GOOGLE_SERVICE_ACCOUNT_ID` array types instead of comma-separated string types as @VivekFitkariwala suggested.

danoswaltCL commented:
@VivekFitkariwala making `GOOGLE_CLIENT_ID` and `GOOGLE_SERVICE_ACCOUNT` strings was to allow for easy transition from one to multiple, without needing to update the value of `GOOGLE_CLIENT_ID` in all existing env configs.

zackcl commented:
@VivekFitkariwala As I checked the `getOsEnvArray` function, it seems like it's still expecting a comma-separated string.

```
export function getOsEnvArray(key: string, delimiter = ','): string[] {
  return (process.env[key] && process.env[key].split(delimiter)) || [];
}
```

Does it mean the `GOOGLE_CLIENT_ID` and `GOOGLE_SERVICE_ACCOUNT_ID` should still remain as they are in the `.env` file, and I just need to update the `AuthService.ts` so that I don't need to do `.split(',');`?

Does that sound correct to you? I originally thought you were suggesting making the variables array types like this:

```
GOOGLE_CLIENT_ID=["id1", "id2", "id3"]
GOOGLE_SERVICE_ACCOUNT_ID=["id1", "id2", "id3"]
```

But looking at the `getOsEnvArray` function, I think you meant that I should simply update the`AuthService.ts` and `env.ts` files. Correct? No changes needed for `.env.example` file. Correct?

Current `.env.example` file:
```
#
# Google Authentication Configuration
# Note: GOOGLE_CLIENT_ID and GOOGLE_SERVICE_ACCOUNT_ID can be a single ID or multiple comma-separated IDs (no spaces)
# Example: id1,id2,id3
#
GOOGLE_CLIENT_ID=google_project_id
GOOGLE_SERVICE_ACCOUNT_ID=google_service_account_id
```

zackcl commented:
@VivekFitkariwala I just made a commit that uses `getOsEnvArray` to parse `GOOGLE_CLIENT_ID` and `GOOGLE_SERVICE_ACCOUNT_ID` into arrays. Please feel free to check if that was what you meant.
